doc/tikv/transaction-handling-newbie-perspective.md

```diff
@@ -0,0 +1,344 @@
+                        ctx.get_isolation_level(),
+                        !ctx.get_not_fill_cache(),
+                        bypass_locks,
+                        false);
+            let result = snap_store.get(&key, &mut statistics)
+                    // map storage::txn::Error -> storage::Error
+                    .map_err(Error::from);
+            result
+        },
+        priority,
+        thread_rng().next_u64(),
+    );
+    res.map_err(|_| Error::from(ErrorInner::SchedTooBusy))
+        .flatten()
+}
+```
+
+This function will get a `snapshot`, and then construct a `SnapshotStore` by using the `snapshot`, and then call `get` on this `SnapshotStore`, and finally get the data we need.
+
+The `bypass_locks` part is a tricky optimize related to [large transaction](https://pingcap.com/blog/large-transactions-in-tidb/), see [this pr](https://github.com/tikv/tikv/pull/5798).
+
+Then we'll view the code of `SnapshotStore::get`, you'll see that in fact it consturcted a [`PointGetter`](https://github.com/tikv/tikv/blob/4ac9a68126056d1b7cf0fc9323b899253b73e577/src/storage/mvcc/reader/point_getter.rs#L133), and then call the `get` method on `PointGetter`:
+
+```rust, no_run
+pub fn get(&mut self, user_key: &Key) -> Result<Option<Value>> {
+    if !self.multi {
+        // Protect from calling `get()` multiple times when `multi == false`.
+        if self.drained {
+            return Ok(None);
+        } else {
+            self.drained = true;
+        }
+    }
+
+    match self.isolation_level {
+        IsolationLevel::Si => {
+            // Check for locks that signal concurrent writes in Si.
+            self.load_and_check_lock(user_key)?;
+        }
+        IsolationLevel::Rc => {}
+    }
+
+    self.load_data(user_key)
+}
+```
+
+As we can see, if the required `isolation_level` is `Si`, we need to check whether there's any locks which may conflict with current get. If we find some, we'll return a  `KeyIsLocked` error:
+
+```rust, no_run
+fn load_and_check_lock(&mut self, user_key: &Key) -> Result<()> {
+    self.statistics.lock.get += 1;
+    let lock_value = self.snapshot.get_cf(CF_LOCK, user_key)?;
+
+    if let Some(ref lock_value) = lock_value {
+        self.statistics.lock.processed += 1;
+        let lock = Lock::parse(lock_value)?;
+        if self.met_newer_ts_data == NewerTsCheckState::NotMetYet {
+            self.met_newer_ts_data = NewerTsCheckState::Met;
+        }
+        lock.check_ts_conflict(user_key, self.ts, &self.bypass_locks)
+            .map_err(Into::into)
+    } else {
+        Ok(())
+    }
+}
+```
+
+And then we'll use `PointGetter`'s `load_data`  method to load the value.
+
+Now we have the value in `GetResponse`, but the client still need to resolve the locked keys. This will still be handled in  `retry_response_stream`.
+
+#### Resolve locks
+
+First, we'll use `take_locks` to take the locks we met, and then we'll use `resolve_locks` to try to resolve them:
```

Comment:
You should clarify that this is the client doing the actions, not TiKV. Also what is `take_locks`?